[Security Solution] Add telemetry for field conflicts in rule updates

jkelas opened the pull request:
### THIS IS STILL A DRAFT!!!!

## Summary

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

miltonhultgren committed:
🌊 Add Group streams behind Advanced Setting (#230265)

This PR is a refined version of
https://github.com/elastic/kibana/pull/219316.
It adds an Advanced Setting for controlling Group Stream availability,
as well as updates the UI with some new elements for Group Streams.

To test you need to call `_enable` and toggle the Advanced Setting.

Advanced Setting:
<img width="1407" height="165" alt="Screenshot 2025-08-12 at 13 26 10"
src="https://github.com/user-attachments/assets/3e51b691-71dd-45b2-a6a3-b6fce1b21945"
/>

Quick video demo:

https://github.com/user-attachments/assets/a6ba6c0b-d33c-486d-b922-627a4e32e6fb

Linked Dashboards view:
<img width="1519" height="974" alt="Screenshot 2025-08-12 at 13 30 58"
src="https://github.com/user-attachments/assets/f3a473c7-bde5-4eeb-89d3-09cd8be0efe8"
/>

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

SoniaSanzV committed:
[Index management] Clean all tests after execution and navigate directly to the tab (#232198)

## Summary
Space time project. I'm trying to improve the test for Kibana
Management. In this PR, I did two things:
* Clean up all created resources after execution. This suite contained a
lot of resources that wasn't deleted after execution. That complicates
local testing and leads to failures if you re-run the tests.
* Navigates directly to the desired tab when setting up the tests. This
is not an anti-patter, but it was time consuming. We were doing this:
```
      await pageObjects.common.navigateToApp('indexManagement');
      // Navigate to the data streams tab
      await pageObjects.indexManagement.changeTabs('data_streamsTab');
      await pageObjects.header.waitUntilLoadingHasFinished();
```

But the `navigateToApp` method already contains a path parameter, so
doing it this way, we can get rid of unnecessaries
`waitUntilLoadingHasFinished` calls.
With this simple change, the execution time of serverless tests locally
dropped from 2 minutes to 1 minute.

* Also, changes the data-test-subj for creating a component template
from `createPipelineButton` to `createComponentTemplateButton`, that
wasn't reflecting what the button really does.

Flay test runner stateful tests:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9177
Flay test runner serverless tests:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9182

alisonelizabeth committed:
add more descriptive a11y test for details flyout (#232300)

Fixes https://github.com/elastic/kibana/issues/219575

While [EUI
recommends](https://eui.elastic.co/docs/components/containers/flyout/#props)
using `aria-labelledby`, in this case, `aria-label` seems more
appropriate as the flyout title consists of only the name.

<img width="690" height="214" alt="Screenshot 2025-08-19 at 12 59 52 PM"
src="https://github.com/user-attachments/assets/0e02dc16-58f6-4a2d-844a-3139217aa094"
/>

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

tobio committed:
Use an explicit string for @timestamp (#233256)

## Summary

Plain YAML strings are not escaped, which means this value is treated as
a literal `\@timestamp` which is not what's intended.
 
Related to https://github.com/elastic/kibana/issues/228077

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

Commit:
WIP

MiriamAparicio committed:
[ObsUX][Infra] Fix Asset details to pick the correct schema instead of default one (#232832)

Co-authored-by: Sergi Romeu <sergi.romeu@elastic.co>

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

NicholasPeretti committed:
[Security solutions] Fix cypress tests (#231341)

## Summary

Closes #198066 

Fixed skipped test.

A couple of test kept crashing cypress and made it quite hard to debug
them. As discussed with @PhilippeOberti these test might become obsolete
in the near future, therefore I've removed them

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: PhilippeOberti <philippe.oberti@elastic.co>

maximpn committed:
[SKA] move generated integration API test clients to a new location (#233112)

**Relates to:** https://github.com/elastic/kibana/pull/231416

## Summary

This PR reconfigures the generation script to output Security Solution auto-generated integration API test clients to the new `x-pack/solutions/security/test/api_integration/services` location.

machadoum committed:
[SecuritySolution] Remove deprecated type (#233290)

## Summary

Remove deprecated type and unused mocks

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine committed:
skip failing test suite (#233044)

elastic-renovate-prod[bot] committed:
Update dependency tough-cookie to v6 (main) (#233120)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [tough-cookie](https://redirect.github.com/salesforce/tough-cookie) |
devDependencies | major | [`^5.1.2` ->
`^6.0.0`](https://renovatebot.com/diffs/npm/tough-cookie/5.1.2/6.0.0) |

---

### Release Notes

<details>
<summary>salesforce/tough-cookie (tough-cookie)</summary>

###
[`v6.0.0`](https://redirect.github.com/salesforce/tough-cookie/releases/tag/v6.0.0)

[Compare
Source](https://redirect.github.com/salesforce/tough-cookie/compare/v5.1.2...v6.0.0)

#### Summary

##### Breaking Changes

- Localhost connections over `http` will now be considered secure by
default. For more information, see the [README
documentation](https://redirect.github.com/salesforce/tough-cookie?tab=readme-ov-file#potentially-trustworthy-origins-are-considered-secure)
and [API
Docs](https://redirect.github.com/salesforce/tough-cookie/blob/master/api/docs/tough-cookie.createcookiejaroptions.md)
for how to configure this feature.

##### Other Notable Changes

-   Dual publishing of ESM+CJS

#### What's Changed

- Bump globals from 15.14.0 to 16.0.0 by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/504](https://redirect.github.com/salesforce/tough-cookie/pull/504)4
- Bump the dev-dependencies group with 10 updates by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/503](https://redirect.github.com/salesforce/tough-cookie/pull/503)3
- Bump tldts from 6.1.76 to 6.1.79 in the production-dependencies group
by [@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/502](https://redirect.github.com/salesforce/tough-cookie/pull/502)2
- Bump tldts from 6.1.83 to 6.1.85 in the production-dependencies group
by [@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/507](https://redirect.github.com/salesforce/tough-cookie/pull/507)7
- Bump the dev-dependencies group with 9 updates by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/508](https://redirect.github.com/salesforce/tough-cookie/pull/508)8
- Bump eslint-import-resolver-typescript from 3.8.3 to 4.3.1 by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/509](https://redirect.github.com/salesforce/tough-cookie/pull/509)9
- feat: Add RFC 6761–compliant localhost loopback checks so secure
cookies work on localhost (fixes:
[#&#8203;382](https://redirect.github.com/salesforce/tough-cookie/issues/382))
by [@&#8203;Chriss4123](https://redirect.github.com/Chriss4123) in
[https://github.com/salesforce/tough-cookie/pull/498](https://redirect.github.com/salesforce/tough-cookie/pull/498)
- use ESM instead of CJS by
[@&#8203;wjhsf](https://redirect.github.com/wjhsf) in
[https://github.com/salesforce/tough-cookie/pull/506](https://redirect.github.com/salesforce/tough-cookie/pull/506)
- Switch from jest to vitest by
[@&#8203;wjhsf](https://redirect.github.com/wjhsf) in
[https://github.com/salesforce/tough-cookie/pull/510](https://redirect.github.com/salesforce/tough-cookie/pull/510)
- Bump vite from 6.2.6 to 6.3.4 by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/521](https://redirect.github.com/salesforce/tough-cookie/pull/521)1
- Bump the dev-dependencies group with 9 updates by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/522](https://redirect.github.com/salesforce/tough-cookie/pull/522)2
- Bump tldts from 6.1.85 to 7.0.5 by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/523](https://redirect.github.com/salesforce/tough-cookie/pull/523)3
- Prepare release v6.0.0-rc.0 by
[@&#8203;colincasey](https://redirect.github.com/colincasey) in
[https://github.com/salesforce/tough-cookie/pull/519](https://redirect.github.com/salesforce/tough-cookie/pull/519)
- Bump the dev-dependencies group with 12 updates by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/525](https://redirect.github.com/salesforce/tough-cookie/pull/525)5
- Bump tldts from 7.0.5 to 7.0.8 in the production-dependencies group by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/524](https://redirect.github.com/salesforce/tough-cookie/pull/524)4
- Create CONTRIBUTING.md by
[@&#8203;wjhsf](https://redirect.github.com/wjhsf) in
[https://github.com/salesforce/tough-cookie/pull/526](https://redirect.github.com/salesforce/tough-cookie/pull/526)
- Bump tldts from 7.0.8 to 7.0.9 in the production-dependencies group by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/530](https://redirect.github.com/salesforce/tough-cookie/pull/530)0
- chore(deps): bump tldts from 7.0.9 to 7.0.10 in the
production-dependencies group by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/532](https://redirect.github.com/salesforce/tough-cookie/pull/532)2
- Bump the dev-dependencies group with 12 updates by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/531](https://redirect.github.com/salesforce/tough-cookie/pull/531)1
- Reverts the check on the Secure attribute when setting a cookie by
[@&#8203;colincasey](https://redirect.github.com/colincasey) in
[https://github.com/salesforce/tough-cookie/pull/534](https://redirect.github.com/salesforce/tough-cookie/pull/534)
- Prepare release v6.0.0-rc.1 by
[@&#8203;colincasey](https://redirect.github.com/colincasey) in
[https://github.com/salesforce/tough-cookie/pull/535](https://redirect.github.com/salesforce/tough-cookie/pull/535)
- Bump the dev-dependencies group with 8 updates by
[@&#8203;dependabot](https://redirect.github.com/dependabot)\[bot]
in[https://github.com/salesforce/tough-cookie/pull/537](https://redirect.github.com/salesforce/tough-cookie/pull/537)7
- Support publishing of both ESM and CJS by
[@&#8203;colincasey](https://redirect.github.com/colincasey) in
[https://github.com/salesforce/tough-cookie/pull/536](https://redirect.github.com/salesforce/tough-cookie/pull/536)
- Prepare v6 by
[@&#8203;colincasey](https://redirect.github.com/colincasey) in
[https://github.com/salesforce/tough-cookie/pull/538](https://redirect.github.com/salesforce/tough-cookie/pull/538)

#### New Contributors

- [@&#8203;Chriss4123](https://redirect.github.com/Chriss4123) made
their first contribution in
[https://github.com/salesforce/tough-cookie/pull/498](https://redirect.github.com/salesforce/tough-cookie/pull/498)

**Full Changelog**:
https://github.com/salesforce/tough-cookie/compare/v5.1.2...v6.0.0

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://redirect.github.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOS4xMDcuMCIsInVwZGF0ZWRJblZlciI6IjM5LjEwNy4wIiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJUZWFtOlNlY3VyaXR5IiwiYmFja3BvcnQ6YWxsLW9wZW4iLCJyZWxlYXNlX25vdGU6c2tpcCJdfQ==-->

Co-authored-by: elastic-renovate-prod[bot] <174716857+elastic-renovate-prod[bot]@users.noreply.github.com>
Co-authored-by: Larry Gregory <larry.gregory@elastic.co>
Co-authored-by: Elena Shostak <165678770+elena-shostak@users.noreply.github.com>

miltonhultgren committed:
🌊 Add Streams global search results provider (#231843)

Adds a new Global search results provider to allow Global search to find
Streams.

<img width="627" height="541" alt="Screenshot 2025-08-14 at 16 31 29"
src="https://github.com/user-attachments/assets/ac159ae3-9f0c-420d-a422-ba63a34929a8"
/>

### How to test

- Enable Streams
- Create a managed Classic Stream
- Create some more child Wired Streams
- Create a Group stream
- Try to search for them in the Kibana global search

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine committed:
skip failing test suite (#231246)

tomsonpl committed:
[EDR Workflows] MDE Cancel - add connector subaction (#233285)

KDKHD committed:
[AI assistant] Add setting to configure default LLM (#231940)

## Summary

Summarize your PR. If it involves visual changes include a screenshot or
gif.

Introduces a new setting in `/app/management/ai/genAiSettings` that
allows users to configure the default LLM. Currently, the setting is
disabled since the changes in the assistants still need to be finalised.
Once the changes in the assistants are completed, the settings can be
made writable.

<img width="3680" height="2382" alt="image"
src="https://github.com/user-attachments/assets/d8842c04-2e98-4e73-b143-cc0458bda564"
/>

<img width="3680" height="2382" alt="image"
src="https://github.com/user-attachments/assets/af890994-b727-4fd8-9f72-1f9b05fb504d"
/>

While we wait for the changes within the assistants, this setting will
be disabled:
<img width="1840" height="1191" alt="image"
src="https://github.com/user-attachments/assets/e7b29c56-8a12-43bb-8b2c-3da8cc4a5a28"
/>

Access to modify these settings is permissioned by the Management >
Advanced Settings in role privileges:

<img width="1840" height="1191" alt="image"
src="https://github.com/user-attachments/assets/9b028b6e-e236-493b-abb0-b95f36a41f32"
/>

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [X] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [X]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [X] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [X] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [X] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [X] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [X] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [X] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

mykolaharmash committed:
[Dataset Quality] Increase table loading timeout in tests (#233340)

Addresses https://github.com/elastic/kibana/issues/232554
Addresses https://github.com/elastic/kibana/issues/232847
Addresses https://github.com/elastic/kibana/issues/232058

In the screenshots from the failed tests above, the table shows in the
loading state and judging from kibana logs, there were no errors on the
backend. It could be the case that infrastructure was particularly slow
and the table would load eventually.

This change increases the timeout to test this assumption and see if
these failures would repeat.

pheyos committed:
Skip serverless search index details test suite for MKI (#233477)

## Summary

This PR skips the serverless search index details test suite for MKI
runs.
Details about the test failure in #233476.

SoniaSanzV committed:
[Ingest Pipeline] Remove test pipeline only if exist (#232487)

Closes https://github.com/elastic/kibana/issues/232442

## Summary

This test only failed once, but it was because we have an after method
tried to delete a pipeline that didn't exist. Wrapping it into a
try/catch will prevent this to fail in the future.

stratoula committed:
[ES|QL] Fixes lookup indexes route failures on read permissions (#233282)

## Summary

This PR is fixing a bunch of bugs all related with lookup join route.

When you are on an instance with read privileges on indexes and try to
use the ES|QL editor you will see 2 problems:

1. The lookup join route fails with 500
2. The failure makes the history component to not save anything at the
local storage. The reason is that our validation api fails so we cant
get the status so the item is not saved at the local storage

After an investigation I found out that:

- client.indices.getSettings fails when you have read priviliges
- the client.indices.getAlias fails if you pass an empty array

What I did:

- changed the client.indices.getSettings to client.indices.resolveIndex
which works on read and doesnt return hidden indices
- I am  fetching the aliases from the same api
- I added the parseMessages to a try catch loop to be sure that even if
something fails on the validation the history component will work

TattdCodeMonkey committed:
[Search] Update homepage promo (#233360)

## Summary

Added new promo for Serverless Elasticsearch.
Abstracted promo components for we can maintain a separate promo for
stateful since it wont be updated as often as stateless.

### Screenshots
Serverless:
<img width="1355" height="824" alt="image"
src="https://github.com/user-attachments/assets/892b082e-f929-4c03-aff6-b4d217e49dc9"
/>

### Checklist

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

machadoum committed:
[SecuritySolution] Add Dataview to Privmon dashboard page (#233264)

## Summary

Add Dataview to Privmon dashboard page.


https://github.com/user-attachments/assets/68cc6488-ef8a-4feb-a080-23cb522f497b

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

paul-tavares committed:
[Security Solution][Endpoint] Complete `runscript` response actions for SentinelOne and show results download link in UI (#232550)

## Summary

The following changes are in support of `runscript` response action for
SentinelOne Host (currently behind feature flag:
`responseActionsSentinelOneRunScriptEnabled`):

- Add logic to check and complete `runscript` response actions for
SentinelOne
- Enabled support for File Info. and File Download APIs for SentinelOne
`runscript`
- Added download link to completed `runscript` action for SentinelOne to
the UI - both in the Response Console as well as in the Response Action
History Log (when row is expanded)

wildemat committed:
[Search][User Education] Shared code examples package and initial api tutorials (#230481)

Closes [#10737 ](https://github.com/elastic/search-team/issues/10737)

## Summary

We want to eventually move current code examples from
[search_indices](https://github.com/wildemat/kibana/blob/e3eee561a4165685e96313b7687c8aaf829c6798/x-pack/solutions/search/packages/kbn-code-examples/src/api-quickstarts/basics_quickstart.test.ts#L10)
to a shared package.

We also want to test out the api quickstart commands in the index
creation flow as outlined in [[User Education] Add Basics quickstart as
run in console
text](https://github.com/elastic/search-team/issues/10321)

## Changes

Changes in this PR can be viewed when enabling the feature flag:
`search_onboarding:searchQuickstartsEnabled`

* This PR creates a shared package for that purpose, and also
initializes the package with console commands taken from [the
docs](https://www.elastic.co/docs/solutions/search/elasticsearch-basics-quickstart).
* Eventually, other `quickstart` guides from the docs can be brought in
to
`x-pack/solutions/search/packages/kbn-code-examples/src/api-quickstarts`.

* Generates a static script that goes into the console. No variables
required for the text content.

* Exposes new quickstart text via "Run In Console" from:
  * Home page, beneath "Connect to Elasticsearch"

## TODO

* ~Decide if we want this out of index flow, and instead as a single CTA
on the homepage~
  * It will eventually be used just from the homepage. Updated as such.
* ~Feature flag both places this is used~
  * Behind feature flag on just the homepage. only place it is used
* ~Finish matching design and layout to the
[mockups](https://www.figma.com/design/m3IT7ReYz8eqeyXSXBtN9q/Onboarding-Code-View-Updates--10414)~
* Addressed some minor feedback from @mdefazio. Removed icons on
quickstarts and got rid of "Show More"
* Depending on where finalize on using this script, add FTR tests
* This change is mostly a placeholder for when we decide to actually
show it to users. Just want the bones there to add more quickstarts
scripts. When we finalize user presentation I think it'll be more
fitting to add tests.
* Copy check w/ docs team
  * See above comment.

## Screenshots

<img width="1137" height="811" alt="Screenshot 2025-08-11 at 11 50
42 AM"
src="https://github.com/user-attachments/assets/f404fad3-da47-4972-b3fc-39fd3cb1c614"
/>


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
~- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials~
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
~- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)~
~- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.~
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
~- [ ] The PR description includes the appropriate Release Notes
section, and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)~
~- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.~

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

e40pud committed:
[Security Solution][AI Assistant] Handle conflicts during the bulk update (#233288)

## Summary

Part of
Epic: https://github.com/elastic/security-team/issues/12768
Meta: https://github.com/elastic/security-team/issues/13657

In my [previous PR](https://github.com/elastic/kibana/pull/232288), I
accidentally removed "retry on conflict"
[logic](https://github.com/elastic/kibana/pull/232288/files#r2305375978).

These changes add a retry logic on bulk update conflicts. For that we
use a `retry_on_conflict` option
https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-bulk#operation-bulk-body-application-json-update-retry_on_conflict.
The retry count is set to 3 to match previous behaviour.

cc @stephmilovic

dokmic committed:
[Core][OnWeek] DI improvements (#232818)

## Summary

This PR adds the following:
- Reduce the number of modules loaded by the DI service.
- Enable autobinding for routes and browser applications to make the
user's code simpler.
- Add `handleLegacyErrors` parameter support to the route definition.
- Encapsulate core services loading to reduce the amount of changes
required when adding a new group of services.
- Add capabilities registry support.
- Add saved objects services support.
- Add logging services support.
- Bump InversifyJS and update the code to adopt new features.

szaffarano committed:
[Telemetry][Indices Metadata] Update index stats query (#233500)

## Summary

Include per-primaries data in index stats query.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

jbudz committed:
update versions.json

logeekal committed:
[Automatic migrations] Dashboard v/s Rule Migration generic logic (#232526)

## Summary

This PR refactors complete Automatic Migrations current data clients and
services for Dashboards and Rules Migrations.

Only objective of this PR is share logic between rules and dashboards
migrations.

### Impact 

This PR is intended to keep the functionality exactly the same. 

### Code Owners Summary

#### elastic/kibana-core

* src/platform/packages/shared/kbn-utility-types/index.ts


#### elastic/security-threat-hunting-investigations

*
x-pack/solutions/security/plugins/security_solution/public/onboarding/components/onboarding_body/cards/siem_migrations/start_migration/start_migration_card.test.tsx


</details>

---------

Co-authored-by: Sergi Massaneda <sergi.massaneda@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

TattdCodeMonkey committed:
search: skip es3 index details error test (flake) (#233509)

## Summary

Skipping flakey test while I diagnose root cause

achyutjhunjhunwala committed:
[Logs UX] Fix codeql and buildkite pipeline issue (#233455)

## Summary

closes -
https://github.com/elastic/kibana-team/issues/1960#issuecomment-3236355328

This PR resolves two critical issues -:

1. Security: [Fixed CodeQL security alerts for incomplete string
sanitisation](https://github.com/elastic/kibana-team/issues/1960#issuecomment-3236355328)
2. [CI/CD: Fixed Docker permission denied errors in Buildkite
pipeline](https://buildkite.com/elastic/kibana-opentelemetry-semantic-conventions-sync/builds/3/steps/table?jid=0198f539-82a8-4fb2-9be9-f6704960a5a8)

  Changes Made

  🔒 Security Fix - String Escaping

- File:
src/platform/packages/shared/kbn-otel-semantic-conventions/src/generate.ts
- Issue: CodeQL flagged incomplete sanitization on lines 31, 32, 33, and
38
- Fix: Updated string escaping to properly handle both backslashes and
single quotes

  🐳 Docker Permission Fix

  - File: .buildkite/scripts/steps/otel_semconv_sync.sh
- Issue: Docker weaver command failing with "Permission denied" when
writing output files
- Root Cause: User ID mapping conflict between host and container
environment
- Solution: Pre-create output directory with proper permissions before
Docker execution

The CI fix is still trial and error. I am discussing with Kibana Ops to
see how to better check for such errors

TattdCodeMonkey committed:
[Search][Homepage] re-arrange sections to move docs above alt solutions (#233503)

## Summary

Moves the docs section above the alternate solution section on the
search homepage

### Screenshot
<img width="1704" height="1192" alt="image"
src="https://github.com/user-attachments/assets/42a74cde-cec6-4dcf-be2c-837164686fc8"
/>

### Checklist

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

e40pud committed:
[Security Solution][AI Assistant] Revert conversation mappings changes (#233428)

## Summary

Epic: https://github.com/elastic/security-team/issues/12768
Meta: https://github.com/elastic/security-team/issues/13657
RFC: [internal
link](https://docs.google.com/document/d/13jAJ5Q3_At_zAuwKjvpYehnM5uzKJSwZIZ1aJJNAf5k)

This PR reverts conversation mappings changes introduced in
https://github.com/elastic/kibana/pull/232288. Since we decided to put
on hold the Using Past Conversation as Context feature, we decided to
remove added summary field backed by the `semantic_text` field to avoid
potential issues that users mights have when updating a conversation
while the inference endpoint is not available.

**Changes**:
* Removed mapping fields: `semantic_content` and
`summarized_message_ids`
* As part of these changes, the `summary` field was hidden on the API
level and conversations response object won't have it anymore

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

jeramysoucy committed:
Adds documentation of baseline screenshotting network policy to disallow external (#233522)

## Summary

Adds an example of a baseline screenshotting network policy to
disallowing all requests to external paths.

jbudz committed:
update versions.json

delanni committed:
Pin Moon to 1.38.6 (#233438)

## Summary
downgrade moon to 1.38.6 - after 1.39+ there's an implicit dependency on
xz (liblzma) on ARM Mac OSX, that is silently failing the build process
- the fix is simple, `brew install xz` but since `moon` is failing with
an exit code 0, we can't see the problem in our bootstrap.

We're safer in 1.38 while we get a fix.

pheyos committed:
Bump serverless ES index node memory to 1.5g (#233516)

## Summary

This PR bumps the memory setting for elasticsearch-serverless docker
image index node from 1g to 1.5g.

This addresses test failures (see #232808 and #232806) that we saw after
https://github.com/elastic/elasticsearch-serverless/pull/4362. The tests
were running with high ES heap usage already (~99%) and this is now
surfaced because new shards are no longer allocated in that scenario.
This memory bump is a quick measure to unblock pipelines and we should
investigate whether we could change the test to run with less resources
in the future.

tobio committed:
Items expects a single field, not a list (#233414)

## Summary

From the
[spec](https://swagger.io/docs/specification/v3_0/data-models/data-types/#arrays)
the `items` field should be a single schema defining the types within an
array. If the array may contain multiple types then the schema should
combine those types via `oneOf`.

Related to https://github.com/elastic/kibana/issues/228077


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

JiaweiWu committed:
[Response Ops] Fix Event Log Flaky E2E Test (#227559)

## Summary

Resolves: https://github.com/elastic/kibana/issues/209911

Fixes a flaky test where the `execute-action` events were not appearing.
This PR changes the test procedure to manually trigger the rule runs
instead of letting it run by itself with a low interval.

### Checklist
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

Commit:
[Security Solution] Fix and unskip API integration tests for preventing non-customizable fields from updating for Prebuilt rule types (#232956)

**Resolves: #195921**

## Summary

I am modifying the skipped tests and unskipping them in MKI.

The first attempt of only relying on removing the call to
`refreshSavedObjectIndices` improved the situation for
`installPrebuiltRules` but another problem was that similar error with
privileges occurred for preparing the assets (reason under the hood
being the same).
The failing periodic pipeline build:
https://buildkite.com/elastic/kibana-serverless-security-solution-quality-gate-rule-management/builds/3265/steps/canvas

That is why I am replacing the logic of installing the rule for test.
Instead of creating assets I am building a zip with our fake rules and
calling the `installFleetPackageByUpload` function, as we already do in
other tests. This allows our test security_detection_engine package to
be installed the `official` way.
**The successful periodic pipeline build**:
https://buildkite.com/elastic/kibana-serverless-security-solution-quality-gate-rule-management/builds/3270/steps/canvas

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

stephmilovic committed:
[Security Assistant] `DocumentsDataWriter` user query fix (#233545)

kibanamachine committed:
[api-docs] 2025-08-30 Daily api_docs build (#233561)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1180

kibanamachine committed:
[api-docs] 2025-08-31 Daily api_docs build (#233566)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1181

SiddharthMantri committed:
chore(login form): Fix layout for Login assistance message (#233497)

## Summary

Closes https://github.com/elastic/kibana/issues/227824

Fixes margin for EUI Horizontal rule component when Login Assistance
Message is present


### Test
Run kibana with the following setting in kibana.(dev.)yaml
```
xpack.security.loginAssistanceMessage: Test Login assistance message
```
On main vs this PR branch, you should see the difference as shown below.


| Main | After fix |
|--------|--------|
| <img width="500" alt="image"
src="https://github.com/user-attachments/assets/ab00606f-aaf2-446f-9118-4f8ce30bf6bb"
/> | <img width="500" alt="Screenshot 2025-08-29 at 15 39 42"
src="https://github.com/user-attachments/assets/0bdc8948-00a5-4b5f-b4fb-52f0ae04709c"
/> |

talboren committed:
feat: Display prominent instructional messages in UI on how to connect the workflow to a rule via the `Run Workflow` action (#233316)

### Add alert trigger notification dots to YAML editor

Adds visual indicators for alert triggers in the workflow YAML editor:
- 🟠 Orange dots appear in the glyph margin next to `type: alert` lines
- Hover tooltips explain alert trigger behavior



https://github.com/user-attachments/assets/d11247de-5b63-40fa-9e47-b0b728809e35

talboren committed:
Workflows Empty State (#233575)

Implemented empty state for workflows list page according to Figma
design using EUI EmptyPrompt component.


<img width="3024" height="1570" alt="CleanShot 2025-08-31 at 13 32
25@2x"
src="https://github.com/user-attachments/assets/882e8e0b-08ed-4b19-b9b9-2b3289fb9cae"
/>

kibanamachine committed:
[api-docs] 2025-09-01 Daily api_docs build (#233587)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1182

tobio committed:
Integers should be type:integer (#233253)

## Summary

From the
[spec](https://swagger.io/docs/specification/v3_0/data-models/data-types/#numbers)
`type:number` defines floating point numbers optionally with `type:
float32|float64`. Integer fields should be `type:integer` where `format`
can be one of `int32` or `int64` if required.

Related to https://github.com/elastic/kibana/issues/228077

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine committed:
skip failing test suite (#233044)

Commit:
Send event for all updates
